android-components/components/support/utils/src/test/java/mozilla/components/support/utils/WebURLFinderTest.kt

```diff
@@ -118,4 +126,101 @@ class WebURLFinderTest {
         assertFalse("content://com.test.app/test".isValidWebURL())
         assertFalse("coNTent://com.test.app/test".isValidWebURL())
     }
+
+    @Test
+    fun isUrlLikeEmulated() {
+        // autolinkWebUrlPattern uses a copy of the regex from URLStringUtils,
+        // so here we emulate isURLLike() and copy its tests.
+        val isURLLike: (String) -> Boolean = {
+            find("random_text $it other_random_text") == it.trim()
+        }
+
+        // All cases that behave differently are annotated with INVERT().
+        val INVERT: (Boolean) -> Boolean = { !it }
```

Comment:
Sorry, I forgot to specify that in my previous comment: we are adding complexity with this annotation-like function, that I think we could replace with clear comments around the tests.

We could replace the use of `assertTrue(INVERT(isURLLike(xxx)))` by `assertFalse(isURLLike(xxx))`. Those asserts could be gathered with a comment above explaining that the ideal regex should have the same result as the one in `UrlStringUtils.kt`, but the current one gives a different result. Also explaining that we         allow it for now, until bug 1685152 is fixed.

What do you think? 

Reply:
> replace the use of assertTrue(INVERT(isURLLike(xxx))) by assertFalse(isURLLike(xxx)). Those asserts could be gathered with a comment

I intentionally kept assertTrue/assertFalse the same as the original tests, and replaced the comments with a programmatic annotation, in order to reduce the cognitive effort required to maintain a second copy.  Your suggestion would *increase* complexity when both functions are considered.

Reply:
I do understand the idea and the value of trying to keep both tests as similar as possible.
Nevertheless, this is starting a new pattern in the codebase, and maybe we should try to follow the existing conventions instead?

Maybe we could get a second opinion from someone else? cc: @jonalmeida

Reply:
Hey! I agree with @titooan, I too would prefer avoiding new patterns. Adding these cases that should pass/fail can be in a separate section with a well-worded comment to explain that these should be have differently when the linked bug is fixed.

> I intentionally kept assertTrue/assertFalse the same as the original tests, and replaced the comments with a programmatic annotation, in order to reduce the cognitive effort required to maintain a second copy. Your suggestion would increase complexity when both functions are considered.

The felt complexity of both solutions are indeed subjective. In cases like this, I would recommend the advice of the code maintainers to decide what is the more preferred way, especially in terms of conventions the codebase follows. 🙂

Reply:
done.